core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -325,17 +323,16 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
             amountToWithdraw = _getCollateral(positionData.rawCollateral);
         }
 
-        _removeCollateral(positionData.rawCollateral, amountToWithdraw);
-        amountWithdrawn = _removeCollateral(rawTotalPositionCollateral, amountToWithdraw);
-
-        // Reset withdrawal request.
-        positionData.withdrawalRequestAmount = FixedPoint.fromUnscaledUint(0);
-        positionData.requestPassTimestamp = 0;
+        // Decrement the sponsor's collateral and global collateral amounts.
+        amountWithdrawn = _decrementCollateralBalances(positionData, amountToWithdraw);
 
-        emit RequestWithdrawalExecuted(msg.sender, amountWithdrawn.rawValue);
+        // Reset withdrawal request by setting withdrawal amount and withdrawal timestamp to 0.
+        _resetWithdrawalRequest(positionData);
 
         // Transfer approved withdrawal amount from the contract to the caller.
         collateralCurrency.safeTransfer(msg.sender, amountWithdrawn.rawValue);
+
+        emit RequestWithdrawalExecuted(msg.sender, amountWithdrawn.rawValue);
```

Comment:
I moved this event to the bottom. before it occurred before the `safeTransfer`

Reply:
I moved it above the `safeTransfer` I'm pretty sure to follow the Checks-Effects-Interaction pattern. Events are technically an effect.

Reply:
I will move it back to be consistent with your change from your PR.